deploy/generate_addon_csi.sh

```diff
@@ -143,11 +150,37 @@ $kubeconfig
   rm -rf ${TARGET_FOLDER}
 }
 
+generate_cloud_config() {
+  echo "########## cloud-config ############"
+  cat ${KUBECFG_FILE_NAME}
+  echo
+  echo "########## cloud-init user data ############"
+  if [[ $OSTYPE == 'darwin'* ]]; then
+          KUBECONFIG_B64=$(base64 -b 0 < "${KUBECFG_FILE_NAME}")
+  else
+          KUBECONFIG_B64=$(base64 -w 0 < "${KUBECFG_FILE_NAME}")
+  fi
+  echo "write_files:
+  - encoding: b64
+    content: ${KUBECONFIG_B64}
+    owner: root:root
+    path: /etc/kubernetes/cloud-config
+    permissions: '0644'"
+  rm -r ${TARGET_FOLDER}
+}
+
 create_target_folder
 create_service_account
 create_rolebinding
 get_secret_name_from_service_account
 extract_ca_crt_from_secret
 get_user_token_from_secret
 set_kube_config_values
-assemble_addon_config
+
+if [[ $SCRIPT_TARGET == "RKE1" ]]; then
+  assemble_addon_config
+elif [[ $SCRIPT_TARGET == "RKE2" || $SCRIPT_TARGET == "K3S" ]]; then
+  generate_cloud_config
+else
```

Comment:
can we validate it using case insensitive or lower-case only

Reply:
Perhaps verifying on line 18 to fail a bad input early without creating those secrets.

Reply:
Updated with both suggestions. Please help to recheck it.